boards/BMS/Src/App/states/App_DriveState.c

```diff
@@ -8,7 +7,20 @@ static void DriveStateRunOnEntry(struct StateMachine *const state_machine)
 {
     struct BmsWorld *         world            = App_SharedStateMachine_GetWorld(state_machine);
     struct BmsCanTxInterface *can_tx_interface = App_BmsWorld_GetCanTx(world);
+    struct Accumulator *      accumulator      = App_BmsWorld_GetAccumulator(world);
+
     App_CanTx_SetPeriodicSignal_STATE(can_tx_interface, CANMSGS_BMS_STATE_MACHINE_STATE_DRIVE_CHOICE);
+
+    // Init the cell state of charge
+    for (uint8_t curr_segment = 0U; curr_segment < NUM_OF_ACCUMULATOR_SEGMENTS; curr_segment++)
```

Comment:
could we move the for loop into the App_initSoc?

Reply:
we dont have a nice way of reading cell voltages. I can pass in a function pointer to do this to the App Soc object, but i rlly dont feel like doing this right now

Reply:
im not using malloc to allocate App_Soc rn since i dont want to mess with unit tests related stuff